src/components/molecules/UiRating/UiRating.vue

```diff
@@ -235,11 +235,11 @@ const itemsToRender = computed<RatingRenderItem[]>(() => (Array.from({ length: m
     ...radioOptionAttrs,
     index: index + 1,
     iconActiveAttrs: {
-      icon: defaultProps.value.settings.iconActive,
+      icon: props.settings.iconActive,
       ...radioOptionAttrs?.iconActiveAttrs,
     },
     iconDefaultAttrs: {
-      icon: defaultProps.value.settings.iconDefault,
+      icon: props.settings.iconDefault,
```

Comment:
IMO this should stay as it is, and the actual fix should be on lines 203-206:

```
    settings: {
      iconDefault,
      iconActive,
    },
```

should be changed to:

```
    settings: ...props.settings,
```